Assignment-5/CarDrive.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    CarAbrupltyStopException,
+    CollisionAccidentException,
+    HitAndRunAccidentException,
+    EnvironmentalAccidentException,
+    CarBrakeFailException
+  } from './exceptions';
+    
+  try {
+    driveCar();
+  } catch (e) {
+    if (e instanceof CarBrakeFailException) {
+      console.log("Car Brake Fail Exception occurred.");
+      handleBreakFailDamage();
+    } else if (e instanceof CollisionAccidentException) {
+      console.log("Collision Accident Exception occurred.");
+      handleCollisionAccident();
+    } else if (e instanceof EnvironmentalAccidentException) {
+      console.log("Environmental Accident Exception occurred.");
+      handleNatureRelatedAccident();
+    } else if (e instanceof HitAndRunAccidentException) {
+      console.log("Hit and Run Accident Exception occurred.");
+      handleHitAndRunAccident();
+    }else if (e instanceof CarAbrupltyStopException){
+      console.log("Car Abruplty Stop Exception occurred.");
+      AbrupltyException();
+    } else {
+      console.error("An error occurred:", e.message);
+    }
+  } finally {
+    cleanUpAfterAccident();
```

Comment:
**cleanUpAfterAccident** method could be more specific about the clean up we are referring to here

Reply:
I don't think we should call **cleanUpAfterAccident** method in finally block, because if there is no exception in try block (i.ie. if there is no accident) then we don't need to do call this method, but in finally block it will always be called.

Reply:
yes I will be making it more specific , irrespective of whether an accident occur or not